DOC: Replace wikipedia links (#4853)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/scim-provisioning-integration-prepare/main/index.md

layout: Guides
---

> Use this guide to prepare a SCIM-compliant (System for Cross-domain Identity Management) API server to host your SCIM service. Then test your server to ensure it's functional.

Reviewer: ```suggestion
Use this guide to prepare a SCIM-compliant (System for Cross-domain Identity Management) API server to host your SCIM service. Then, test your server to ensure that it's functional.
```

## Preparation

The first step in delivering your SCIM integration is preparing a SCIM-compliant API server to host your SCIM service.

Okta supports lifecycle provisioning using either version 2.0 and version 1.1 of the SCIM protocol.

> If your service already supports the SCIM protocol, you should still review the [Okta SCIM reference](/docs/reference/scim/) documentation. They help you to understand how Okta implements the SCIM protocol.

Reviewer: ```suggestion
If your service already supports the SCIM protocol, you should still review the [Okta SCIM reference](/docs/reference/scim/) documentation. The SCIM reference helps you to understand how Okta implements the SCIM protocol.
```

### API endpoints

> The API endpoint for your SCIM API must be secured using the Transport Layer Security (TLS) protocol. Connections through this secure layer are routed by using the `https://` prefix for your URL. See the [MDN article on Transport Layer Security](https://developer.mozilla.org/en-US/docs/Web/Security/Transport_Layer_Security).

Author: ```suggestion
The API endpoint for your SCIM API must be secured using the [Transport Layer Security (TLS)](https://developer.mozilla.org/en-US/docs/Web/Security/Transport_Layer_Security) protocol. Connections through this secure layer are routed by using the `https://` prefix for your URL.
```

Okta supports authentication against SCIM APIs using any one of the following methods:

- OAuth 2.0 [Authorization Code grant flow](https://tools.ietf.org/html/rfc6749#section-4.1)

> - [Basic authentication (MDN article)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#basic_authentication_scheme)

Author: ```suggestion
- [Basic authentication](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#basic_authentication_scheme)
```

## SCIM facade

Sometimes it isn't feasible for your cloud-based application to natively support a SCIM API. An alternative is to build and host your own SCIM facade middleware. It can translate between the Okta SCIM API connection and the cloud application's proprietary API. The Okta integration connection is then made to this SCIM facade.

## Provision to on-premises applications

> This provisioning guide targets cloud-based apps. For on-premises apps, see [Provision on-premises provisioning](https://help.okta.com/okta_help.htm?id=ext_OPP_configure) for details about the Okta agent-based provisioning solution.

Reviewer: I checked and the name of the page has changed, thankfully - because Provision on-premises provisioning is just nuts for a title haha
```suggestion
This provisioning guide targets cloud-based apps. For on-premises apps, see [Provision on-premises applications](https://help.okta.com/okta_help.htm?id=ext_OPP_configure) for details about the Okta agent-based provisioning solution.
```

    | idUserOne | `323da8f8-21b8-4b25-8322-90673d9e1bc7` | A UUID of a test user created on your SCIM server. |
    | randomUserEmail | `Runscope300Hfluaklab151@example.com` | An email address returned from your SCIM server. |

> 1. To see the details of tests, click the name of each particular test case to expand the section. The details have information on the **Request**, **Response**, and **Connection** for each HTTP request involved in the test. Each test sends a composed GET or POST request to your SCIM server. The HTTP status received in response determines the success or failure of each test.

Reviewer: ```suggestion
1. To see test details, click the name of each particular test case to expand the section. The details have information on the **Request**, **Response**, and **Connection** for each HTTP request involved in the test. Each test sends a composed GET or POST request to your SCIM server. The HTTP status received in the response determines the success or failure of each test.
```

> 1. After the test is completed, the main panel displays the results of your test.

Reviewer: ```suggestion
1. After the test finishes, the main panel displays the test results.
```

> Since this test is running in your own Runscope instance, you can modify the tests to better fit your own environment. Then complete the test run it again. If you need more technical details, see the [SCIM Reference](/docs/reference/scim/), or check out [Test your SCIM API](/docs/guides/scim-provisioning-integration-prepare/main/#test-your-scim-api).

Reviewer: Is it "complete the test run again" or "run the test run again"?

```suggestion
Since this test is running in your own Runscope instance, you can modify the tests to better fit your own environment. Then, complete the test run again. If you need more technical details, see the [SCIM Reference](/docs/reference/scim/), or check out [Test your SCIM API](/docs/guides/scim-provisioning-integration-prepare/main/#test-your-scim-api).
```
